sign_language_datasets/datasets/dgs_corpus/dgs_corpus.py

```diff
@@ -222,75 +283,124 @@ def _split_generators(self, dl_manager: tfds.download.DownloadManager):
+
+    def _include_videos(self, datum: Any, features: Any):
+        videos = {}
+        for participant in ["a", "b", "c"]:
+            video_key = "video_" + participant
+
+            if video_key not in datum.keys() or datum[video_key] is None:
+                video = ""
+            else:
+                video = str(datum[video_key])
+
+            videos[participant] = video
+
+        # make sure that the video fps is as expected
+        for video_path in videos.values():
+            if video_path == "":
+                continue
+
+            cap = cv2.VideoCapture(video_path)
+            actual_video_fps = cap.get(cv2.CAP_PROP_FPS)
+            cap.release()
 
-    def _generate_examples(self, data):
+            assert math.isclose(actual_video_fps, float(DEFAULT_FPS)), "Framerate of video '%s' is %f instead of %d" % (
+                video_path,
+                actual_video_fps,
+                DEFAULT_FPS,
+            )
+
+        features["fps"] = self._builder_config.fps if self._builder_config.fps is not None else DEFAULT_FPS
+        features["paths"]["videos"] = videos
+
+    def _generate_examples(self, data, split: List[str] | Dict[str, List[str]] = None):
         """ Yields examples. """
 
-        default_fps = 50
         default_video = np.zeros((0, 0, 0, 3))  # Empty video
 
         for _id, datum in list(data.items()):
+            if split is not None and _id not in split:
+                continue
+
             features = {
                 "id": _id,
                 "paths": {t: str(datum[t]) if t in datum else "" for t in ["ilex", "eaf", "srt", "cmdi"]},
             }
 
             if self._builder_config.include_video:
-                videos = {}
-                for participant in ["a", "b", "c"]:
-                    video_key = "video_" + participant
-
-                    if video_key not in datum.keys() or datum[video_key] is None:
-                        video = ""
-                    else:
-                        video = str(datum[video_key])
-
-                    videos[participant] = video
-
-                # make sure that the video fps is as expected
-
-                for video_path in videos.values():
-                    if video_path == "":
-                        continue
-
-                    cap = cv2.VideoCapture(video_path)
-                    actual_video_fps = cap.get(cv2.CAP_PROP_FPS)
-                    cap.release()
-
-                    assert math.isclose(actual_video_fps, float(default_fps)), "Framerate of video '%s' is %f instead of %d" % (
-                        video_path,
-                        actual_video_fps,
-                        default_fps,
-                    )
-
-                features["fps"] = self._builder_config.fps if self._builder_config.fps is not None else default_fps
-                features["paths"]["videos"] = videos
+                self._include_videos(datum, features)
+
+            poses = None
```

Comment:
Because of this line I believe I get an error. Using this code:

```python
from sign_language_datasets.datasets.dgs_corpus.dgs_corpus import DgsCorpusConfig

config = DgsCorpusConfig(name="only-annotations", version="1.0.0", include_video=False, include_pose=None)
dgs_corpus = tfds.load('dgs_corpus', builder_kwargs=dict(config=config))
```

I currently get

```python
RuntimeError: KeyError: Failed to encode example:
{'id': '1413451-11105600-11163240', 'paths': {'ilex': '/root/tensorflow_datasets/downloads/sign-lang.uni-hamb.de_mein_ilex_1413-1110OPo7Tex6TwPc65yWsuEieyUb1JRiDF8s6dol7X-B-D8.ilex', 'eaf': '/root/tensorflow_datasets/downloads/sign-lang.uni-hamb.de_mein_eaf_1413-1110-1CjuYhPq5xoPeLIzOWSv0PUHLTddtCdBHXL2YWIDOsZ4.eaf', 'srt': '/root/tensorflow_datasets/downloads/sign-lang.uni-hamb.de_mein_srt_1413-1110-1pk40s6YBh23ak5SqPJTWbiHFN1cDK0O0ipK0DI1EueI.srt', 'cmdi': '/root/tensorflow_datasets/downloads/sign-lang.uni-hamb.de_mein_cmdi_1413-11103lQv7mavXxzAx9O9BHAy4H1vUgIyTtpLF0bpPdM2iHQ.cmdi'}, 'poses': None}
'poses'
```

because if poses are not requested in the config, I think that "poses" should not be added as an entry in the feature dict.

Reply:
fixed. quite an oversight on my part.... manually tested all sentence configurations, and not all document ones